### ISSUE TITLE
Metricbeat 7.5.2 causes old dockerd versions to hang or provided slow responses

### ISSUE THREAD

**Issue by eedugon** [elastic/beats#17240]
Metricbeat (running in a docker) is not able to fetch metrics from `docker` module due to dockerd being hanged. However, using the exact same configuration with Metricbeat 7.5.1 it works fine and never fails.

The symptoms are easy to see: when metricbeat 7.5.2 is started, docker commands are very slow some times, like a simple `docker ps` taking 10 or 15 seconds. When Metricbeat 7.5.1 is running the same commands are always fast / instant.

The errors that appear when docker daemon is not responsive from Metricbeat point of view are:
```
2020-02-19T13:26:00.367+0100	INFO	module/wrapper.go:252	Error fetching data for metricset docker.memory: failed to get docker stats: Cannot connect to the Docker daemon at unix:///hostfs/var/run/docker.sock. Is the docker daemon running?
2020-02-19T13:26:26.369+0100	ERROR	elasticsearch/client.go:404	Failed to encode event: unsupported float value: NaN
```

This is happening RHEL 7.7 which comes with a bit old docker daemon release:
__Host OS__: Red Hat Enterprise Linux Server release 7.7 (Maipo), kernel 3.10.0-1062.el7.x86_64
__Docker version__: docker-1.13.1-103.git7f2769b.el7.x86_64

**Comment by eedugon** [comment]
The change introduced in 7.5.2 that causes this is:
https://github.com/elastic/beats/pull/15224

As we can see, adding `?size=1` to the API call makes the responses to be much slower:
```
# time curl -i  --unix-socket /var/run/docker.sock http://v1.24/containers/json > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15605    0 15605    0     0  10.3M      0 --:--:-- --:--:-- --:--:-- 14.8M
 
real    0m0.008s
user    0m0.002s
sys     0m0.005s


# time curl -i  --unix-socket /var/run/docker.sock http://v1.24/containers/json?size=1 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15964    0 15964    0     0    512      0 --:--:--  0:00:31 --:--:--  5106
 
real    0m31.163s
user    0m0.003s
sys     0m0.010s
```

This is not happening with recent releases of docker daemon

**Comment by eedugon** [comment]
The following issue has created to revert the mentioned PR back:
https://github.com/elastic/beats/issues/16601

Backports:
7.x elastic/beats#16617
7.6 elastic/beats#16619

For the moment this has been solved by reverting the 7.5.2 change in 7.6, 7.7 and 8.x
